internal/ssh/fingerprint.go

```diff
@@ -0,0 +1,82 @@
+package ssh
+
+import (
+	"crypto/md5" //#nosec
+	"crypto/x509"
+	"encoding/hex"
+	"fmt"
+	"strings"
+
+	"golang.org/x/crypto/ssh"
+)
+
+// OpenSSHFingerprints is the de-facto standard OpenSSH fingerprints for SSH public keys:
+// SHA256 (used for ED type keys) and MD5 (used for RSA keys). Fingerprints are returned as
+// string encoded into base64 or hex respectively. Additionally, type and comment are also
+// returned. Type as one of the: "ssh-ed25519" or "ssh-rsa".
+type OpenSSHFingerprints struct {
+	Type    string
+	SHA256  string
+	MD5     string
+	Comment string
+}
+
+// AWSFingerprint is a special Amazon EC2 MD5 fingerprint calculated from the PEM-encoded
+// public key. It is used in AWS for ssh keys.
+type AWSFingerprint string
+
+//#nosec
+func md5Separator(data []byte, separator byte) string {
+	hashSlice := md5.Sum(data)
+	var sb strings.Builder
+	buf := make([]byte, 2)
+	sb.Grow(47)
+	for i := 1; i <= len(hashSlice); i++ {
+		hex.Encode(buf, hashSlice[i-1:i])
+		sb.Write(buf)
+		if i < 16 {
+			sb.WriteByte(separator)
+		}
+	}
+	return sb.String()
+}
+
+// GenerateOpenSSHFingerprints parses a public key and returns OpenSSH fingerprints.
+func GenerateOpenSSHFingerprints(pubkeyBody []byte) (OpenSSHFingerprints, error) {
+	fps := OpenSSHFingerprints{}
+
+	pkey, cmt, _, _, err := ssh.ParseAuthorizedKey(pubkeyBody)
+	if err != nil {
+		return OpenSSHFingerprints{}, fmt.Errorf("unable to parse public key %s: %w", pubkeyBody, err)
+	}
+
+	fps.Comment = cmt
+	fps.Type = pkey.Type()
+	fps.SHA256 = strings.TrimLeft(ssh.FingerprintSHA256(pkey), "SHA256:") + "="
```

Comment:
Why is `=`  added here?

Reply:
that's just a padding `=` is an empty character included to keep a constant length, which usually is there in fingerprints, but Golang library is not including it for some reason.

Reply:
Base64 can be either padded or not padded, AWS requires padding.

https://linuxhint.com/understand-base64-padding/

Reply:
Thanks @ezr-ondrej and @lzap, confirms we're doing the 'right' thing on our end by stripping this padding when present.